### PR TITLE
Update .travis.yml

### PR DESCRIPTION
**Pull request by mmatera**
Inkscape can now be installed from the standard repository

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -57,8 +57,8 @@ before_install:
     fi
   - if [[ "$DOC" == "true" ]]; then
       sudo apt-get install -qq asymptote texlive-latex-extra texlive-latex3 texlive-fonts-recommended;
-      sudo apt-get install -qq  texlive-font-utils latexmk texlive-xetex lmodern;
-      sudo add-apt-repository -y ppa:inkscape.dev/stable-daily && sudo apt-get update && sudo apt-get install -qq inkscape;
+      sudo apt-get install -qq  texlive-font-utils latexmk texlive-xetex lmodern inkscape;
+      # sudo add-apt-repository -y ppa:inkscape.dev/stable-daily && sudo apt-get update && sudo apt-get install -qq inkscape;
```
Comment (GarkGarcia) [inline .travis.yml:61]: We should delete this before merging the PR.

**Comment by rocky** [comment]
Going to Focal and 3.8 by default gives a [failure in running PyPy 3.6](https://travis-ci.org/github/rocky/Mathics/jobs/726760550). Anyone care to investigate?

**Comment by rocky** [comment]
> Let's disable this for a while to see if  compiles OK without it

Yes for now sadly that is what I am coming to think too. 

I intend at some point to spend some time to see what is wrong with graphs.  I hope to get this fixed before the release though.

**Comment by mmatera** [comment]
I think That it would be a good candidate to develop and deploy as a external module

**Comment by GarkGarcia** [comment]
> I think That it would be a good candidate to develop and deploy as an external module

Agreed. I depends on `graphviz` (which is not required for anything else) and it tends to make things harder in testing side.

**Comment by rocky** [comment]
> I think That it would be a good candidate to develop and deploy as an external module

Ok. I understand how to create a Mathematica Package (I did this for Combinatorica). Is this what you mean or something different. If something different do you have pointers on how to do. Or would you like to do this?

**Comment by mmatera** [comment]
The new version includes LoadModule functionality, and as an example, I moved natlang from builtin to be a  pymathics module. You can use it as an example. This functionality requires a better documentation, but I think we could write it along the road...

**Comment by GarkGarcia** [comment]
>   error: library mach has Fortran sources but no Fortran compiler found

[This](https://travis-ci.org/github/mathics/Mathics/jobs/726773459#L1186) is interesting. LOL

**Comment by GarkGarcia** [comment]
> ! LaTeX Error: File `asymptote.sty' not found.
> 
> Type X to quit or <RETURN> to proceed,
> or enter new name. (Default extension: sty)
> 
> Enter file name: 

[This](https://travis-ci.org/github/mathics/Mathics/jobs/726773460#L6627) is also interesting. First of all, we need to get `asymptote` installed in VM. Also, we should pass the `-halt-on-error` flat to prevent this kind of from happening. This build will never complete, baucase `xelatex` is expecting us to tell it where `asymptote.sty` is.

**Comment by GarkGarcia** [comment]
> First of all, we need to get asymptote installed in VM.

@mmatera You mentioned the tests worked on your machine. Which TexLive version do you use? Also, which of `texlive-` packages have you installed?

It may be a pain to install `asymptote` via `tlmgr`, because `tlmgr` only runs in Debian systems when it's up-to-date (and most `texlive-` packages are not).

**Comment by GarkGarcia** [comment]
> > First of all, we need to get asymptote installed in VM.
> 
> @mmatera You mentioned the tests worked on your machine. Which TexLive version do you use? Also, which of `texlive-` packages have you installed?
> 
> It may be a pain to install `asymptote` via `tlmgr`, because `tlmgr` only runs in Debian systems when it's up-to-date (and it's usually impossible to get it up-to-date via `apt-get`).

**Comment by mmatera** [comment]
> @mmatera

In my laptop, I have Ubunto 18( Bionic Beaver) with these texlive packages:

texlive/                   texlive-extra-utils/       texlive-latex-base/        texlive-plain-generic/     
texlive-base             texlive-fonts-recommended texlive-latex-extra       texlive-pstricks
texlive-binaries          texlive-font-utils        texlive-latex-recommended texlive-publishers        
texlive-doc               texlive-lang-spanish      texlive-pictures          texlive-xetex

Maybe we could test travis with Bionic, to see if everything fixes alone...

**Comment by GarkGarcia** [comment]
> Maybe we could test travis with Bionic, to see if everything fixes alone...

I don't think this is a good idea. We need to find out in which of the `texlive-` packages `asymptote` is contained, This is most likelly the cause of the bug.

**Comment by GarkGarcia** [comment]
For reference: https://asymptote.sourceforge.io/doc/LaTeX-usage.html

@mmatera Is the `aymptote` package getting installed in CI?

**Comment by GarkGarcia** [comment]
> For reference: https://asymptote.sourceforge.io/doc/LaTeX-usage.html
> 
> @mmatera Is the `asymptote` package getting installed in CI?

Try adding `asymptote` to the `apt-get` line.

**Comment by mmatera** [comment]
The asymptote latex package is supposed to be installed by the asymptote Deb package...

**Comment by mmatera** [comment]
It was, for the documentation section...

**Comment by GarkGarcia** [comment]
> It was, for the documentation section...

Yes, I'm sorry, just checked it out. This is very weird.

**Comment by mmatera** [comment]
I mean, let's give a try with Bionic, and then, if it is an issue of Focal, then we can tell about it to the mantainers of the asymptote package...

**Comment by GarkGarcia** [comment]
> I mean, let's give a try with Bionic, and then, if it is an issue of Focal, then we can tell about it to the mantainers of the asymptote package...

Sure, let's try it.

**Comment by GarkGarcia** [comment]
I'd consider https://travis-ci.org/github/mathics/Mathics/jobs/726820131#L16248 to be progress.

**Comment by GarkGarcia** [comment]
> I'd consider https://travis-ci.org/github/mathics/Mathics/jobs/726820131#L16248 to be progress. Also, things are working again in PyPy.

**Comment by mmatera** [comment]
With bionic, everything is in green! let's try to pass it to focal


> Also, things are working again in PyPy.

**Comment by GarkGarcia** [comment]
> With bionic, everything is in green!

The `DOC=true` tests are still failing unfortunately ☹️ (see https://travis-ci.org/github/mathics/Mathics/jobs/726823828#L6592). Here is re relevant error message:

```
Traceback (most recent call last):
  File "test.py", line 330, in <module>
    main()
  File "test.py", line 314, in main
    write_latex()
  File "test.py", line 253, in write_latex
    with open_ensure_dir(settings.DOC_TEX_DATA, "rb") as output_file:
  File "test.py", line 181, in open_ensure_dir
    return open(f, *args, **kwargs)
FileNotFoundError: [Errno 2] No such file or directory: '/home/travis/build/mathics/Mathics/mathics/doc/tex/data'
```

It looks like someone forgot to add a `'+'` at the end of the second argument of `open`. In other words, people are calling `open(f, 'w')` instead of `open(f, 'w+')`. The `'+'` here indicates that Python is supposed to create the file if it doesn't exist already.

I'm not sure if your intention is to fix it before we merge this PR.

**Comment by mmatera** [comment]
mmm...
this one worked:
https://travis-ci.org/github/mmatera/Mathics/builds/726823809

but with focal failed again...

**Comment by GarkGarcia** [comment]
> mmm...
> this one worked:

That's very weird indeed. To be quite honest with you, I got no clue on why this things are failing 😅️

**Comment by mmatera** [comment]
Regarding the error, it is funny because it belongs to a file that I didn't touch in the branch... I just was working on .travis and the mathics.tex file (apart from commenting out the graphics module)

**Comment by GarkGarcia** [comment]
> Regarding the error, it is funny because it belongs to a file that I didn't touch in the branch... I just was working on .travis and the mathics.tex file (apart from commenting out the graphics module)

Exactly. This hasn't even been changed, I'm not sure how it could cause an error.

**Comment by mmatera** [comment]
OK, I made a rebase now to tidy up the changes. Let's see if now it works..

**Comment by mmatera** [comment]
The fails seem to come from the new implementation of the tests in  #884. I think it would be safe to accept this PR. When #884 be ready, everything should work fine together, assuming we use bionic as the base distribution.

**Comment by rocky** [comment]
Please feel free to revise the Makefile and/or add to #884 thanks. I seem to recall you saying you like modularity. This is an example of modularity.

**Comment by rocky** [comment]
> > Regarding the error, it is funny because it belongs to a file that I didn't touch in the branch... I just was working on .travis and the mathics.tex file (apart from commenting out the graphics module)
> 
> Exactly. This hasn't even been changed, I'm not sure how it could cause an error.

My guess is that there were a number of errors there, but we just weren't seeing them either because the tests didn't test that aspect, or because things were done in a way that overwrite verus append doesn't make a difference.

**Comment by mmatera** [comment]
Could you merge this PR? this would ensure that packages for the documentation tests be properly installed, and then, we could deal with the bugs from the make system of Mathics.

**Comment by GarkGarcia** [comment]
> @GarkGarcia, would you mind merge this PR? this would ensure that packages for the documentation tests be properly installed, and then, we could deal with the bugs from the make system of Mathics.

No problem! I wasn't sure if you considered it to be finished already.

**Comment by GarkGarcia** [comment]
@mmatera Just to make sure we're on the same page, we haven't got the tests for `graphs` working again, correct?

**Comment by rocky** [comment]
My plan was to have it appear as a module in pymathics. When the testing of modules is fixed so only they test only what is in the moidule, it will be easier to work on this.

**Comment by GarkGarcia** [comment]
> My plan was to have it appear as a module in pymathics. When the testing of modules is fixed so that what is in the module only, it will be easier to work on this

Makes sense.

**Pull request merged**